doc/concepts.qbk

```diff
@@ -199,6 +199,73 @@ the generator, for example to re-run a test suite at a later time.]
+
+[note Quasi-random sequences are known to give efficient numerical integration 
+rules in many Bayesian statistical problems where the posterior distribution can be 
+transformed into periodic functions on a hypercube.]
+
+Harold Niederreiter gives an extensive overview on random number generation
+and quasi-Monte Carlo methods in his book "Random number generation and
+quasi-Monte Carlo methods, Society for Industrial and Applied Mathematics, 1992". 
+
+In addition to the __UniformRandomNumberGenerator requirements,
+a quasi-random number generator has some additional requirements. In the
+following table, `X` denotes a quasi-random number generator class, `u` is a value of `X`, 
+`v` is a const value of `X`, and `j` a value of type `unsigned long long`.
+
+[table QuasiRandomNumberGenerator requirements
+  [[expression] [return type] [pre/post-condition]]
+  [[`X(s)`] [-] [creates an s-dimensional generator with a default seed. Dimension `s` is an integer no less than 1.]]
+  [[`v.dimension()`] [std::size_t] [the dimension of quasi-random domain.]]
+  [[`u.seed(i)`] [`void`] [seeds the generator with the integer `i`.]]
+  [[`u.discard(j)`] [`void`] [Advances the generator by `j` steps as if by
+                              `j` calls to `u()`.]]
+ ]
+
+[note The `operator()` returns a successive element of an s-dimensional (`s` = `v.dimension()`) vector 
+at each invocation. When all elements are exhausted, `operator()` begins anew with the starting
+element of a subsequent s-dimensional vector.]
+
+Classes which model a quasi-random number generator shall also model
+__EqualityComparable, i.e. implement `operator==`. Two quasi-random number
+generators are defined to be /equivalent/ if they both return an identical
+sequence of numbers starting from a given state.
+
+Classes which model a quasi-random number generator shall also model the
+__Streamable concept, i.e. implement `operator<<` and `operator>>`.
+`operator<<` writes all current state of the quasi-random number generator
+to the given `ostream` so that `operator>>` can restore the state at a later
+time. The state shall be written in a platform-independent manner, but it is
+assumed that the `locales` used for writing and reading be the same. The
+quasi-random number generator with the restored state and the original at
+the just-written state shall be equivalent.
+
+Classes which model a quasi-random number generator should also model the
+__CopyConstructible and __Assignable concepts. However, note that the
+sequences of the original and the copy are strongly correlated (in fact,
+they are identical), which may make them unsuitable for some problem domains.
+Thus, copying quasi-random number generators is discouraged; they should
+always be passed by (non-const) reference.
+
+The classes __niederreiter_base2, __sobol, __faure are models for a quasi-random number generator.
+
+[endsect]
+
 [section Seed Sequence]
 
 A SeedSeq represents a sequence of values that can be used to
```

Comment:
Should the documentation give a list of known discrepancy bounds for each sequence?

Reply:
Should the docs discuss how quasi-random sequences exploit low effective dimension? See "Necessity of Low Effective Dimension" by Art Owen.

Reply:
1) Sounds ok.
2) I think, umm... no. I think we should try and maintain some balance between this being a simple documentation of the API, and a full-blown introduction to the finer nuances of QRNGs.